File: VRPC#/ConsoleApplication1/ConsoleApplication1/approaches/IntervalVarComparator_w.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApplication1.approaches
{

    using IloIntervalVar = ilog.concert.IloIntervalVar;

    public class IntervalVarComparator_w : IComparer<IloIntervalVar>
    {
        public virtual int Compare(IloIntervalVar arg0, IloIntervalVar arg1)
        {
            if (arg1.SizeMax < arg0.SizeMax)
            {
                return -1;
            }
            if (arg1.SizeMax > arg0.SizeMax)
            {
                return 1;
            }
            return 0;
        }
    }

}

File: 2L-HFVRP/src/approaches/IntervalVarComparator_w.java
package approaches;

import ilog.concert.IloIntervalVar;

import java.util.Comparator;

public class IntervalVarComparator_w implements Comparator<IloIntervalVar>
{
	public int compare(IloIntervalVar arg0, IloIntervalVar arg1) 
	{
		if (arg1.getSizeMax() < arg0.getSizeMax()) return -1;
		if (arg1.getSizeMax() > arg0.getSizeMax()) return 1;
		return 0;
	}
}

